server/src/resources/js/cart.ts

```diff
@@ -0,0 +1,194 @@
+import productTemplate from './templates/productTemplate';
+import errorTemplate from './templates/errorTemplate';
+import { injectable, inject } from 'inversify';
+import { GernzyGraphqlService } from './interfaces/graphqlService';
+import { TYPES } from './types/types';
+import { StoreProducts } from './interfaces/products';
+import { GernzyCart } from './interfaces/cart';
+
+@injectable()
+class Cart implements GernzyCart {
+    private graphqlService: GernzyGraphqlService;
+    private productObj: StoreProducts;
+    private url: string = '';
+
+    public constructor(
+        @inject(TYPES.GernzyGraphqlService) private _graphqlService: GernzyGraphqlService,
+        @inject(TYPES.StoreProducts) private _productObj: StoreProducts,
+    ) {
+        this.graphqlService = _graphqlService;
+        this.productObj = _productObj;
+    }
+
+    public endpointUrl(url: string) {
+        this.url = url;
+    }
+
+    public cartSetup() {
+        var userToken = localStorage.getItem('userToken') || '';
+        var self = this;
+
+        window.cartProducts = () => {
+            return {
```

Comment:
Can we de-nest this somewhat, maybe as another private method, just to keep the code structure as flat as possible. e.g.

`window.cartProducts = this.createPublicInterface()`

Reply:
completed, just had to figure out why the scope of 'this' didn't work, and had to use .bind(this)